### ISSUE TITLE
gnome-keyring required to stay logged in

### ISSUE THREAD

**Issue by AeliusSaionji** [flathub/com.skype.Client#20]
Would it be possible to supply gnome-keyring as part of this package? Skype will not stay logged in without it.

**Comment by TingPing** [comment]
I might go the other way. Chromium itself allows setting different keyring types by passing `--password-store=basic`, I wonder if Skype exposes this in any way.

**Comment by bam80** [comment]
Any progress? Having Skype logged out every time it closes is pretty annoying.

**Comment by AeliusSaionji** [comment]
Simply install gnome-keyring yourself in the meantime.

**Comment by snaggen** [comment]
Doesn't Skype use libsecret ? If so, then kwallet should work also.

**Comment by TingPing** [comment]
Unless my information is outdated, KDE never implemented the secret service API.

**Comment by snaggen** [comment]
Maybe this should be part of some xdg-desktop-portal? So, that when something access the keyring api, it could detect that there is no keyring installed and inform the user. And is this really unique for the flatpak version of Skype? Wouldn't any other skype installation behave the same way if you do not have a keyring installed?

**Comment by TingPing** [comment]
> And is this really unique for the flatpak version of Skype? Wouldn't any other skype installation behave the same way if you do not have a keyring installed?

You are correct.

> Maybe this should be part of some xdg-desktop-portal? 

The future I, and some others, want is where there is no such thing as a Secret Service. It is a concept that doesn't fit with sandboxing. There is a `libsecret` PR somewhere to just store keys locally.

**Comment by zdenek-zikan** [comment]
Any progress on this or workaround? I am using KDE, but have gnome-keyring-daemon running (installed by the OS package manager (dnf)). Now I installed Skype from Flatpak and it keeps asking  for gnome-keyring password but nothing works. Skype installed from rpm doesn't have this problem.

**Comment by WhyNotHugo** [comment]
Giving skype access to the host secretservice kinda defeats the purpose of isolation, since secretservice provides no form of authorisation: it simply gives anyone access to all passwords.

That said, the current setup _does_ give Skype access to the host secretservice, and it still doesn't work.

**Comment by TingPing** [comment]
Current releases of libsecret no longer require access to the host service closing that hole.

**Comment by WhyNotHugo** [comment]
@TingPing Can you explain how this works? I see that the flatpak has "talk" access to `org.freedesktop.secrets`; is this access filtered somehow by some part of flatpak?

As far as I can see, skype itself tries to use the secrets API. It fails on Flatpak, but works on Firejail. However, with Firejail it gets unrestricted access to the secrets service, which is not nice at all.

I also tried using bubblewrap + xdg-dbus-proxy directly, and that also works, but, again, giving full access to the secret service.

#131 is strongly related here, and, as per my testing, Skype fails to read from the keyring on Flatpak: https://github.com/flathub/com.skype.Client/issues/131#issuecomment-1080837974

**Comment by WhyNotHugo** [comment]
@TingPing Can you explain how this works? I see that the flatpak has "talk" access to org.freedesktop.secrets; is this access filtered somehow by some part of flatpak?

As far as I can see, skype itself tries to use the secrets API. It fails on Flatpak, but works on Firejail. However, with Firejail it gets unrestricted access to the secrets service, which is not nice at all.

I also tried using bubblewrap + xdg-dbus-proxy directly, and that also works, but, again, giving full access to a secret service.

https://github.com/flathub/com.skype.Client/issues/131 is strongly related here, and, as per my testing, Skype fails to read from the keyring on Flatpak: https://github.com/flathub/com.skype.Client/issues/131#issuecomment-1080837974

**Comment by TingPing** [comment]
> @TingPing Can you explain how this works? I see that the flatpak has "talk" access to org.freedesktop.secrets; is this access filtered somehow by some part of flatpak?

No. I'm not sure if it works and this permission might bypass it. `libsecret` itself is supposed to support being ran in a sandbox transparently.
